[BUG] Generator creates invalid code on validated schema when an enum contains ~=

#### Bug Report Checklist

- [X] Have you provided a full/minimal spec to reproduce the issue?
- [X] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [X] What's the version of OpenAPI Generator used?
- [X] Have you search for related issues/PRs?
- [X] What's the actual output vs expected output?
- [ ] [Optional] Bounty to sponsor the fix ([example](https://www.bountysource.com/issues/66123212-javascript-client-produces-a-wrong-object-for-a-string-enum-type-that-is-used-with-ref))

<!--
Please follow the issue template below for bug reports.
Also please indicate in the issue title which language/library is concerned. Eg:  [BUG][JAVA] Bug generating foo with bar 
-->

##### Description

<!-- describe what is the question, suggestion or issue and why this is a problem for you. -->

A schema for Jira validates but cannot be used for generation (crashing with "Please make sure that your schema is correct!"). This worked with the last 3.X.Y release, but I'd like to use 4.X.Y as it has some fixes for Rust bindings.

##### openapi-generator version

<!-- which version of openapi-generator are you using, is it a regression? -->
4.1.3

It's a regression that appears to have been present since 4.0. Before that did not have the same problem.

##### OpenAPI declaration file content or url

<!-- if it is a bug, a json or yaml that produces it.
If you post the code inline, please wrap it with
```yaml
(here your code)
```
(for YAML code) or
```json
(here your code)
```
(for JSON code), so it becomes more readable. If it is longer than about ten lines,
please create a Gist (https://gist.github.com) or upload it somewhere else and
link it here.
  -->

https://developer.atlassian.com/cloud/jira/platform/swagger-v3.v3.json 

##### Command line used for generation

<!-- including the language, libraries and various options -->
```
java -jar openapi-generator-cli.jar generate -i jira-v3.swagger-v3.json -g rust -o abc
```

##### Steps to reproduce

<!-- unambiguous set of steps to reproduce the bug.-->
```
$ cd $(mktemp -d)
$ VER=4.1.3 && wget http://central.maven.org/maven2/org/openapitools/openapi-generator-cli/$VER/openapi-generator-cli-$VER.jar -O openapi-generator-cli.jar
[...]
$ curl -sSL https://developer.atlassian.com/cloud/jira/platform/swagger-v3.v3.json > jira-v3.swagger-v3.json
$ java -jar openapi-generator-cli.jar validate -i jira-v3.swagger-v3.json --recommend
Validating spec (jira-v3.swagger-v3.json)
Warnings: 
        -Unused model: RichText
        -Unused model: JqlQueryUnitaryOperand
        -Unused model: FieldValueClause
        -Unused model: JqlQueryClauseTimePredicate
        -Unused model: avatarUrls
        -Unused model: ListOperand
        -Unused model: IconBean
        -Unused model: CompoundClause
        -Unused model: FieldChangedClause
        -Unused model: KeywordOperand
        -Unused model: JqlQueryClauseOperand
        -Unused model: ValueOperand
        -Unused model: FunctionOperand
        -Unused model: FieldWasClause

[info] Spec has 14 recommendation(s).
$ rm -rf abc && mkdir abc && java -jar openapi-generator-cli.jar generate -i jira-v3.swagger-v3.json -g rust -o abc
[main] INFO  o.o.c.ignore.CodegenIgnoreProcessor - No .openapi-generator-ignore file found.
[main] INFO  o.o.codegen.DefaultGenerator - OpenAPI Generator: rust (client)
[main] INFO  o.o.codegen.DefaultGenerator - Generator 'rust' is considered stable.
[main] WARN  o.o.codegen.utils.ModelUtils - Multiple schemas found in content, returning only the first one
[main] WARN  o.o.codegen.utils.ModelUtils - Multiple schemas found in content, returning only the first one
[main] WARN  o.o.codegen.utils.ModelUtils - Multiple schemas found in content, returning only the first one
[main] WARN  o.o.codegen.utils.ModelUtils - Multiple schemas found in content, returning only the first one
[main] WARN  o.o.codegen.DefaultCodegen - Unknown type found in the schema: object
[main] WARN  o.o.codegen.DefaultCodegen - Unknown type found in the schema: object
[main] INFO  o.o.codegen.DefaultGenerator - Model JqlQueryClause not generated since it's a free-form object
[main] INFO  o.o.codegen.DefaultGenerator - Model ConnectModule not generated since it's a free-form object
[main] INFO  o.o.codegen.DefaultGenerator - Model ConnectModules not generated since it's an alias to map (without property) and `generateAliasAsModel` is set to false (default)
[main] INFO  o.o.codegen.DefaultGenerator - Model ListWrapperCallbackApplicationRole not generated since it's a free-form object
Exception in thread "main" java.lang.RuntimeException: Could not process model 'FieldValueClause'.Please make sure that your schema is correct!
        at org.openapitools.codegen.DefaultGenerator.generateModels(DefaultGenerator.java:472)
        at org.openapitools.codegen.DefaultGenerator.generate(DefaultGenerator.java:923)
        at org.openapitools.codegen.cmd.Generate.run(Generate.java:416)
        at org.openapitools.codegen.OpenAPIGenerator.main(OpenAPIGenerator.java:60)
Caused by: java.lang.NullPointerException
        at org.openapitools.codegen.DefaultCodegen.updateCodegenPropertyEnum(DefaultCodegen.java:4304)
        at org.openapitools.codegen.DefaultCodegen.postProcessModelsEnum(DefaultCodegen.java:385)
        at org.openapitools.codegen.languages.RustClientCodegen.postProcessModels(RustClientCodegen.java:156)
        at org.openapitools.codegen.DefaultGenerator.processModels(DefaultGenerator.java:1233)
        at org.openapitools.codegen.DefaultGenerator.generateModels(DefaultGenerator.java:467)
        ... 3 more
```

##### Related issues/PRs

<!-- has a similar issue/PR been reported/opened before? Please do a search in https://github.com/openapitools/openapi-generator/issues?utf8=%E2%9C%93&q=is%3Aissue%20 -->
I see a number of issues mentioning [updateCodegenPropertyEnum](https://github.com/OpenAPITools/openapi-generator/search?q=updateCodegenPropertyEnum&type=Issues) but nothing recent or relating to a crash, and a few mentioning "Please make sure that your schema is correct", but that seems to be a generic error.

##### Suggest a fix

<!-- if you can't fix the bug yourself, perhaps you can point to what might be
  causing the problem (line of code or commit), or simply make a suggestion -->

I note that it's crashing on a model that is recommended for removal, so I'll try that and see if it helps.

--- a/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java
+++ b/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java
@@ -1558,6 +1558,7 @@ public class DefaultCodegen implements CodegenConfig {
         specialCharReplacements.put("<=", "Less_Than_Or_Equal_To");
         specialCharReplacements.put(">=", "Greater_Than_Or_Equal_To");
         specialCharReplacements.put("!=", "Not_Equal");
+        specialCharReplacements.put("~=", "Tilde_Equal");
     }
 
     /**

